src/olympia/versions/models.py

```diff
@@ -231,10 +231,15 @@ def get_url_path(self):
             return ''
         return reverse('addons.versions', args=[self.addon.slug, self.version])
 
-    def delete(self):
+    def delete(self, hard=False):
         log.info(u'Version deleted: %r (%s)' % (self, self.id))
         amo.log(amo.LOG.DELETE_VERSION, self.addon, str(self.version))
-        super(Version, self).delete()
+        if (hard):
+            super(Version, self).delete()
+        else:
+            # Soft delete
```

Comment:
I think it's clear that we're not actually deleting here so maybe it'd be better to add to the comment why this feature exists rather than describing the two lines below.